kernel/Utilities/BasisFunctions1DH1ConformingLine.cpp

```diff
@@ -72,17 +72,27 @@ double BasisFunction1DInteriorLine::evalDeriv0(
                LobattoPolynomialDerivative(polynomialOrder_, p[0]) / 4.;
 }
 
-Base::BasisFunctionSet* createDGBasisFunctionSet1DH1Line(
+std::vector<Base::BaseBasisFunction*> createDGBasisFunctions1DH1Line(
     std::size_t polynomialOrder) {
-    Base::BasisFunctionSet* result(new Base::BasisFunctionSet(polynomialOrder));
+    std::vector<Base::BaseBasisFunction*> result;
     if (polynomialOrder > 0) {
-        result->addBasisFunction(new BasisFunction1DVertexLine(0));
-        result->addBasisFunction(new BasisFunction1DVertexLine(1));
+        result.emplace_back(new BasisFunction1DVertexLine(0));
+        result.emplace_back(new BasisFunction1DVertexLine(1));
         for (std::size_t i = 0; i + 2 <= polynomialOrder; ++i) {
-            result->addBasisFunction(new BasisFunction1DInteriorLine(i));
+            result.emplace_back(new BasisFunction1DInteriorLine(i));
         }
     } else {
-        addPiecewiseConstantBasisFunction1D(*result);
+        result.emplace_back(createPiecewiseConstant1D());
+    }
+    return result;
+}
+
+Base::BasisFunctionSet* createDGBasisFunctionSet1DH1Line(
+    std::size_t polynomialOrder) {
+
+    Base::BasisFunctionSet* result(new Base::BasisFunctionSet(polynomialOrder));
+    for (auto* bf : createDGBasisFunctions1DH1Line(polynomialOrder)) {
+        result->addBasisFunction(bf);
     }
     return result;
 }
```

Comment:
we could start introducing unique_ptrs

Reply:
That might result in quite a bit of refactoring. As that would impact all the basis functions, as these are passed around to the mesh. Preferably I would redesign this basis function stuff at some point.

All the changes in this file and the one for H1ConformingTriangle, are just so that I can reuse these basis functions as building block (the modulator) for the 2D Nedelec ones. So I preferably do as little as possible in these files.

Reply:
yes okay